sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1347,6 +1347,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val DECIMAL_OPERATIONS_MINIMUM_ADJUSTED_SCALE =
+    buildConf("spark.sql.decimalOperations.minimumAdjustedScale")
+      .internal()
+      .doc("Decimal operations' minimum adjusted scale when " +
```

Comment:
Can we improve this description explaining clearly what this means? A user may be confused if he/she is not familiar.

Reply:
@mgaido91 how about this:
```
Decimal arithmetic operations' minimum adjusted scale in rounding the decimal part of result. When spark.sql.decimalOperations.allowPrecisionLoss is true and when a result precision is greater than MAX_PRECISION (38), the corresponding scale is reduced to prevent the integral part of a result from being truncated. This conf controls how many decimal places to keep at minimum.
```

Reply:
I think this is still hacky and it does not resolve the root issue. Let us think about it and see whether we have a better solution. I change the target version to 3.0. We will revisit this before the code freeze of 3.0. If we do not have a solution, we can review this again.

Reply:
I am not sure which is the problem experienced for which this PR was opened. I think this value can be made a config, as there may be specific use cases where this value may be tuned properly, despite I can't think of specific ones.

> it does not resolve the root issue.

What do you mean by "the root issue"? AFAIK, there is only one main issue with decimal operations rigth now, which is a division when a decimal with negative scale is involved. For this problem, #22450 is waiting for reviews, after positive feedback on that approach in the discussion in the mailing list.